librz/core/cbin.c

```diff
@@ -2848,6 +2848,8 @@ static int bin_sections(RzCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 	const char *type = print_segments ? "segment" : "section";
 	bool plugin_type_support = plugin && plugin->section_type_to_string;
 	bool plugin_flags_support = plugin && plugin->section_flag_to_rzlist;
+	bool is_elf = info->rclass && !strcmp(info->rclass, "elf");
+	bool is_mach0 = info->rclass && !strcmp(info->rclass, "mach0");
```

Comment:
Hardcoding formats is a bad practice, imo just iterate through `sections` first and check if there is alignment info in one entry. Then you also don't need the `info` at all.

Reply:
Yes, I had thought about checking all the sections first but I wonder whether it will be more slow. Now I only know two kinds of formats having `align` info. I think file formats will be relatively stable so we are not likely to add new `align` frequently.

But to avoid hardcoding, I can iterate through `sections` first. And do you think I should keep the `info` to be `NULL`?

Reply:
Since iterating over all sections is necessary for printing anyway, the complexity will not change, so we don't need to worry much about performance here.

If `info` is always `NULL` and thus effectively unused, it can be removed completely. It might be a leftover after refactoring.